### PR TITLE
Implement ChangeCompanyColour game command

### PR DESCRIPTION
**Pull request by AaronVanGeffen**
Split off from #1028

### CODE REVIEW

File: src/OpenLoco/Map/Tile.h [hunk 1]
```diff
@@ -115,8 +115,6 @@ namespace OpenLoco::Map
             return type() == TClass ? (TType*)this : nullptr;
         }
 
-        uint8_t baseZ() const { return _base_z; }
```
Comment (AaronVanGeffen) [inline src/OpenLoco/Map/Tile.h:118]: This _private_ `TileElement` member function was shadowing a _public_ `TileElementBase` member function.

File: src/OpenLoco/Company.cpp [hunk 2]
```diff
@@ -131,4 +133,65 @@ namespace OpenLoco
...
+                }
+            }
+        }
+        Gfx::invalidateScreen();
+    }
+
+    // 0x0042F0C1
+    static void updateHeadquartersColourAtTile(const Map::TilePos2& pos, uint8_t zPos, Colour_t newColour)
+    {
+        auto tile = Map::TileManager::get(pos);
```
Comment (duncanspumpkin) [inline src/OpenLoco/Company.cpp:169]: Really we should be invalidating the screen at each of these tiles. ATM it doesn't matter as we invalidate the whole screen in the vehicle update colour function that is always called before this but it was just a thought i had.

**Review by duncanspumpkin — APPROVED** [review]

File: src/OpenLoco/Company.cpp (hunk 2 above)
Comment (AaronVanGeffen) [reply src/OpenLoco/Company.cpp:169]: That's a good point. Currently, this function is only called by the ChangeCompanyColour game command, so it won't be a problem, as you said. Any preference either way?

Comment (duncanspumpkin) [reply src/OpenLoco/Company.cpp:169]: not really just an observation.

**Pull request merged**